src/scripts/components/tipping/TipOverlay.js

```diff
@@ -0,0 +1,176 @@
+/* @flow */
+
+import React from 'react'
+import styled from 'styled-components'
+
+import ParatiiLib from 'utils/ParatiiLib'
+import { TOKEN_UNITS } from 'constants/ParatiiLibConstants'
+
+import { VIDEO_OVERLAY_PADDING } from 'constants/UIConstants'
+import {
+  TIPPING_UI_STEPS,
+  MAXIMUM_LAST_SECURED_FOR_PASSWORD_PROMPT_SECONDS
+} from 'constants/TippingConstants'
+
+import Video from 'records/VideoRecords'
+
+import TranslatedText from 'components/translations/TranslatedText'
+import BackButton from 'components/foundations/buttons/BackButton'
+import CloseButton from 'components/foundations/buttons/CloseButton'
+import Colors from 'components/foundations/base/Colors'
+import ChooseAmountTipStep from './steps/ChooseAmountTipStep'
+import ConfirmTipStep from './steps/ConfirmTipStep'
+import TipCompleteStep from './steps/TipCompleteStep'
+
+import type { TippingUIStep } from 'types/TippingTypes'
+
+type Props = {
+  addDoNotTipVideo: (videoId: string) => void,
+  balancesAreLoading: boolean,
+  ptiBalance: string,
+  lastSecuredTimestamp: number,
+  loadBalances: () => void,
+  notification: (Object, string) => void,
+  setUserIsTipping: (isTipping: boolean) => void,
+  video: Video
+}
+
+type State = {
+  currentStep: TippingUIStep,
+  tipAmount: number
+}
+
+const Wrapper = styled.div`
+  width: 100%;
+  height: 100%;
+  background-color: ${Colors.black};
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  align-items: center;
+  text-align: center;
+  position: relative;
+`
+
+const BackButtonWrapper = styled.div`
+  position: absolute;
+  top: ${VIDEO_OVERLAY_PADDING};
+  left: ${VIDEO_OVERLAY_PADDING};
+`
+
+const CloseButtonWrapper = styled.div`
+  position: absolute;
+  top: ${VIDEO_OVERLAY_PADDING};
+  right: ${VIDEO_OVERLAY_PADDING};
+`
+
+class TipOverlay extends React.Component<Props, State> {
+  constructor (props: Props) {
+    super(props)
+
+    this.state = {
+      currentStep: TIPPING_UI_STEPS.CHOOSE_AMOUNT,
+      tipAmount: 0
+    }
+
+    this.props.loadBalances()
+  }
+
+  backToChooseAmountStep = () => {
+    this.setState({
+      currentStep: TIPPING_UI_STEPS.CHOOSE_AMOUNT
+    })
+  }
+
+  transferTip = async () => {
+    try {
+      const tipAmountWei: number = Number(
+        ParatiiLib.eth.web3.utils.toWei(`${this.state.tipAmount}`)
+      )
+      await ParatiiLib.eth.transfer(
```

Comment:
@jellegerbrandy I just added a `toWei` conversion above, but I'm still seemingly able to transfer any amount to anyone. Going to debug some more.